lms/static/js/spec/edxnotes/views/tabs/recent_activity_spec.js

```diff
@@ -50,28 +70,135 @@ define([
                 'templates/edxnotes/note-item', 'templates/edxnotes/tab-item'
             ]);
 
-            this.collection = new NotesCollection(notes);
+            this.collection = new NotesCollection(notes, {perPage: 10, parse: true});
             this.tabsCollection = new TabsCollection();
         });
 
         it('displays a tab and content with proper data and order', function () {
             var view = getView(this.collection, this.tabsCollection);
+            Helpers.verifyPaginationInfo(view, "Showing 1-3 out of 3 total", 1, 1);
+            Helpers.verifyPageData(view, this.tabsCollection, tabInfo, recentActivityTabId, notes);
+        });
 
-            expect(this.tabsCollection).toHaveLength(1);
-            expect(this.tabsCollection.at(0).toJSON()).toEqual({
-                name: 'Recent Activity',
-                identifier: 'view-recent-activity',
-                icon: 'fa fa-clock-o',
-                is_active: true,
-                is_closable: false,
-                view: 'Recent Activity'
-            });
-            expect(view.$('#recent-panel')).toExist();
-            expect(view.$('.note')).toHaveLength(3);
-            _.each(view.$('.note'), function(element, index) {
-                expect($('.note-comments', element)).toContainText(notes[index].text);
-                expect($('.note-excerpt', element)).toContainText(notes[index].quote);
-            });
+        it("will not render header and footer if there are no notes", function () {
+            var notes = Helpers.createNotesData(
+                {
+                    numNotesToCreate: 0,
```

Comment:
Why we are calling this method with `numNotesToCreate = 0` ? I think we don't need this call.